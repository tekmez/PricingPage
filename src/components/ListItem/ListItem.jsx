import React from 'react'
import { View, Text } from 'react-native'
import styles from './ListItem.style'
const ListItem = ({ icon, title, description }) => {
  return (
    <View testID="list-item-container" style={styles.container}>
      <View testID="list-item-icon" style={styles.iconContainer}>
        {icon}
      </View>
      <View testID="list-item-info" style={styles.infoContainer}>
        <Text testID="list-item-title" style={styles.title}>
          {title}
        </Text>
        <Text testID="list-item-description" style={styles.description}>
          {description}
        </Text>
      </View>
    </View>
  )
}

export default ListItem
