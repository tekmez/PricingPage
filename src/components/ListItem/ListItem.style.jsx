import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 20,
    gap: 16,
  },
  iconContainer: {
    width: 40,
    height: 40,
    borderRadius: 8,
    borderColor: '#E7E8E9',
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  infoContainer: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  title: {
    fontWeight: 'bold',
    fontSize: 14,
    lineHeight: 20,
  },
  description: {
    fontSize: 12,
    lineHeight: 16,
    fontFamily: 'Inter-Regular',
    fontWeight: '400',
    color: '#717578',
  },
})
export default styles
