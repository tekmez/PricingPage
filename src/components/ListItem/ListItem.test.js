import React from 'react'
import { render } from '@testing-library/react-native'
import ListItem from './ListItem'

describe('List Item', () => {
  it('should match snapshot', () => {
    const iconName = 'test-icon'
    const title = 'Test Title'
    const description = 'Test Description'
    const { toJSON } = render(
      <ListItem icon={iconName} title={title} description={description} />
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render title correctly', () => {
    const testTitle = 'Test Title'
    const comp = render(<ListItem title={testTitle} />)
    const titleText = comp.getByTestId('list-item-title').children[0]
    expect(titleText).toBe(testTitle)
  })

  it('should render description correctly', () => {
    const testDescription = 'Test Description'
    const comp = render(<ListItem description={testDescription} />)
    const descriptionText = comp.getByTestId('list-item-description')
      .children[0]
    expect(descriptionText).toBe(testDescription)
  })

  it('should render icon correctly', () => {
    const iconName = 'test-icon'
    const comp = render(<ListItem icon={iconName} />)
    const icon = comp.getByTestId('list-item-icon').children[0]
    expect(icon).toBe(iconName)
  })
})
