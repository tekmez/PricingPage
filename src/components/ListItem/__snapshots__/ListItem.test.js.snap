// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 16,
      "paddingVertical": 20,
    }
  }
  testID="list-item-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#E7E8E9",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 40,
        "justifyContent": "center",
        "width": 40,
      }
    }
    testID="list-item-icon"
  >
    test-icon
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
    testID="list-item-info"
  >
    <Text
      style={
        {
          "fontSize": 14,
          "fontWeight": "bold",
          "lineHeight": 20,
        }
      }
      testID="list-item-title"
    >
      Test Title
    </Text>
    <Text
      style={
        {
          "color": "#717578",
          "fontFamily": "Inter-Regular",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 16,
        }
      }
      testID="list-item-description"
    >
      Test Description
    </Text>
  </View>
</View>
`;
